index.js

```diff
@@ -0,0 +1,60 @@
+
+var canvas = document.querySelector("canvas"), 
+ctx = canvas.getContext("2d");
+
+function draw(n,data){
+    let width = n, 
+    height = n, 
+    scale = canvas.width / n; 
+    fetch(data)
+        .then(response => response.json())
+        .then(cdata => {
+            
+            let makeColor = function(d) {return d;};
+            
+            if(data == 'data/4x4.json'){
+                makeColor = function(d){ return '#' + d; };
+            }else if(data == 'data/32x32.json'){
+                makeColor = function(d){ 
+                 let [r,g,b,a] = d;
+                    return `rgba(${r}, ${g}, ${b}, ${a})`; 
+                };
+            }
+
+          
+             for(var row = 0; row < height; row++) {
+                 for(var col = 0; col < width; col++) {
+                    ctx.fillStyle = makeColor(cdata[col][row]);
+                    ctx.fillRect(col * scale, row * scale, scale, scale);
+                }
+            }
+         });       
+}
+ function drawPNG(){
+    const img = new Image(canvas.width, canvas.height);
+    img.onload = ()=>{
+        ctx.drawImage(img, 0, 0);
+    }
+    img.src = 'data/image.png';
+}
```

Comment:
also I'd suggest putting functions at the end of the file as common approach for hoisting.